pkg/ccl/sqlproxyccl/tenant/directory.proto

```diff
@@ -129,10 +129,14 @@ message Tenant {
   string cluster_name = 3;
   // ConnectivityType indicates the extent to which the tenant can be accessed.
   ConnectivityType connectivity_type = 4;
+  // PublicCIDRRanges corresponds to the list of source CIDR ranges that are
+  // allowed to access the tenant. This will only apply if the tenant allows
+  // public connections.
+  repeated string public_cidr_ranges = 5 [(gogoproto.customname) = "PublicCIDRRanges"];
```

Comment:
I don't think we should link any of these concepts to what's in CC. "Public CIDR ranges" is what the proxy trying to enforce. In the context of Kubernetes, we use `LoadBalancerSourceRanges`. I'd like something that's generic here. How do you feel about `allowed_cidr_ranges`, `allowed_ranges`, or `source_ranges`? It would be nice to have the term `public` though, to distinguish itself from `private_endpoints`. If we had it has `ip_entries`, I would expect that to be actual IP addresses, and not CIDR ranges.

As for supporting SQL and UI, the same argument can be applied to private endpoints as well. For now, we will assume that both are for SQL only. Once we start supporting UI through the "SQL proxy" (which I think is unlikely [1]), we can revisit this decision. Note that these names are not encoded anywhere since they are just protobuf messages, so we have a lot of flexibility here.

[1] We would have more problems here since we're calling it a _SQL_ proxy.

Reply:
I don't know that I necessarily want to link it to the same name just because it's in CC, but an allowlist is something people are familiar with. Like if someone saw the name "Public CIDR ranges", they would have no idea what that is. And if they asked you, you'd probably say "Oh, that's for the IP Allowlist". I really like the name `allowed_cidr_ranges` though, that's a lot more clear what it's there for!

[1] Haha yeah I think we might just call it the Serverless Proxy or something, but it seems like the most natural place to do it, since it already knows how to find tenants and route traffic to them. But that's a decision for another time.

Reply:
Done. I also renamed PrivateEndpoints to AllowedPrivateEndpoints.